utils/SystemSetup.ts

```diff
@@ -116,34 +119,57 @@ export async function setUpFunctionalSystem(
     wbtcDecimals = 18;
 
     // swap btc eth price
-    const euroOldInitialWethPrice = euroInitialWethPrice;
-    euroInitialWethPrice = euroInitialWbtcPrice;
-    euroInitialWbtcPrice = euroOldInitialWethPrice;
-
-    const usdOldInitialWethPrice = usdInitialWethPrice;
-    usdInitialWethPrice = usdInitialWbtcPrice;
-    usdInitialWbtcPrice = usdOldInitialWethPrice;
+    const oldInitialWethPrice = initialStablePrice.ETH;
+    initialStablePrice.ETH = initialStablePrice.BTC;
+    initialStablePrice.BTC = oldInitialWethPrice;
   } else {
     wethDecimals = 18;
     wbtcDecimals = 8;
 
-    euroInitialWethPrice = (await getOnChainEthEurPrice(hre)).price;
-    euroInitialWbtcPrice = (await getOnChainBtcEurPrice(hre)).price;
+    // Get stable prices from price feed
+    await Promise.all(
+      getAllBDStablesFiatSymbols().map(async symbol => {
+        const onChainEthStablePrice = symbol === "USD" ? await getOnChainWethUsdPrice(hre) : (await getOnChainEthStablePrice(hre, symbol)).price;
+        _.set(initialStablePrice, ["ETH", symbol], onChainEthStablePrice);
 
-    usdInitialWethPrice = await getOnChainWethUsdPrice(hre);
-    usdInitialWbtcPrice = await getOnChainWbtcUsdPrice(hre);
+        const onChainBtcStablePrice = symbol === "USD" ? await getOnChainWbtcUsdPrice(hre) : (await getOnChainBtcStablePrice(hre, symbol)).price;
+        _.set(initialStablePrice, ["BTC", symbol], onChainBtcStablePrice);
+      })
+    );
   }
 
   if (hre.network.name === "mainnetFork") {
     verboseLog(verbose, "provide liquidity bdus/usdc");
 
     const usdc = (await hre.ethers.getContractAt("IERC20", constants.EXTERNAL_USD_STABLE[hre.network.name].address)) as IERC20;
-    const amount = 100;
-
-    await provideLiquidity(hre, treasury, bdUs, usdc, to_d18(amount), numberToBigNumberFixed(amount, 6), verbose);
-  }
+    const usdAmount = 100;
```

Comment:
maybe we should make this name more specific? is it only for the xau and gbp?

Reply:
Did you refer to `usdAmount`? if so, it's only for BDUS-USDC on mainnetFork  and I've moved it to constants